language-configuration.json

```
@@ -24,5 +24,7 @@
         ["(", ")"],
         ["\"", "\""],
         ["'", "'"]
-    ]
+    ],
+    // defines what is considered to be a word in the programming language
+    "wordPattern": "(-?\\d*\\.\\d\\w*)|([^\\`\\~\\!\\@\\#\\%\\^\\&\\*\\(\\)\\_\\=\\+\\[\\{\\]\\}\\\\\\|\\;\\:\\'\\\"\\,\\.\\<\\>\\/\\?\\s]+)"
```

I think this will fix your underscore issue

```suggestion
    "wordPattern": "(-?\\d*\\.\\d\\w*)|([^\\`\\~\\!\\@\\#\\%\\^\\&\\*\\(\\)\\=\\+\\[\\{\\]\\}\\\\\\|\\;\\:\\'\\\"\\,\\.\\<\\>\\/\\?\\s]+)"
```

---

I think I should have phrased the description differently. The underscore behavior was intended, since hoon does not allow for words like `my_variable_name` (to the best of my knowledge), but does allow for `_my-variable-name` where `_` isn't part of the word itself.

Let me know if you think it should behave differently.

---

What about this? It would still technically match _some_var but it would break those up into two words, so if hoon doesn't even support that then we should be fine.

```suggestion
    "wordPattern": "(-?\\d*\\.\\d\\w*)|(_?[^\\`\\~\\!\\@\\#\\%\\^\\&\\*\\(\\)\\_\\=\\+\\[\\{\\]\\}\\\\\\|\\;\\:\\'\\\"\\,\\.\\<\\>\\/\\?\\s]+)"
```

---

I took another look at the `hoon.tmLanguage` and realized we could use a simpler pattern that's already used there for arms, auras, etc.

How about this?
```suggestion
    "wordPattern": "[a-z]([a-z0-9-]*[a-z0-9])?"
```